Dockerfile.frontend

```diff
@@ -1,21 +1,41 @@
-FROM node:lts as builder
+# syntax = docker/dockerfile:1.2
+# Base user uid / gid keep 1000 on prod, align with your user on dev
+ARG USER_UID=1000
+ARG USER_GID=1000
 
-RUN mkdir -p /opt/product-opener/node_modules
-COPY package*.json /opt/product-opener/
-COPY .snyk /opt/product-opener/
+FROM node:lts as builder
+ARG USER_UID
+ARG USER_GID
+RUN usermod --uid $USER_UID node && \
+    groupmod --gid $USER_GID node && \
+    mkdir -p /opt/product-opener/node_modules && \
+    mkdir -p /opt/product-opener/html/data && \
+    mkdir -p /opt/product-opener/html/css/dist && \
+    mkdir -p /opt/product-opener/html/images/icons/dist && \
+    mkdir -p /opt/product-opener/html/js/dist && \
+    mkdir -p /opt/product-opener/html/images/products && \
+    chown node:node -R /opt/product-opener/
+COPY --chown=node:node package*.json /opt/product-opener/
+COPY --chown=node:node .snyk /opt/product-opener/
 WORKDIR /opt/product-opener
-RUN npm install
+USER node
+RUN --mount=type=cache,id=npm-cache,target=/root/.npm npm install
 ENV PATH /opt/product-opener/node_modules/.bin:$PATH
 
-COPY html /opt/product-opener/html
-COPY icons /opt/product-opener/icons
-COPY scss /opt/product-opener/scss
-COPY gulpfile.js /opt/product-opener/
+COPY --chown=node:node html /opt/product-opener/html
+COPY --chown=node:node icons /opt/product-opener/icons
+COPY --chown=node:node scss /opt/product-opener/scss
+COPY --chown=node:node gulpfile.js /opt/product-opener/
+COPY --chown=node:node .snyk /opt/product-opener/
```

Comment:
Why do we need the `--chown=node:node` in those `COPY` commands ?

Reply:
This is mostly due to snyk which will try to modify libraries on the fly (this is its role) so we need node user to have write capability.